lib/reporters/tap.js

```diff
@@ -143,22 +141,22 @@ function createProducer(tapVersion) {
  * @private
  * @constructor
  */
-function TAPProducer() {}
+class TAPProducer {
 
 /**
  * Writes the TAP version to reporter output stream.
  *
  * @abstract
  */
-TAPProducer.prototype.writeVersion = function () {};
+writeVersion() {};
 
 /**
  * Writes the plan to reporter output stream.
  *
  * @abstract
```

Comment:
Tiny scope creep: the function isn't actually `abstract`, the docs have been wrong this whole time, let's fix it while we're here

```suggestion
```